pow-migration/src/monitor/mod.rs

```diff
@@ -45,23 +47,43 @@ pub async fn get_ready_txns(
     client: &Client,
     validator: String,
     block_window: Range<u32>,
+    pos_genesis_config_hash: &Blake2bHash,
 ) -> Vec<TransactionDetails> {
     if let Ok(transactions) = client.get_transactions_by_address(&validator, 10).await {
+        let genesis_config_hex = pos_genesis_config_hash.to_hex();
+
         let filtered_txns: Vec<TransactionDetails> = transactions
             .into_iter()
-            .filter(|txn| {
-                // Here we filter by current epoch
-                block_window.contains(&txn.block_number)
-                    && txn.to_address == Address::burn_address().to_user_friendly_address()
-                    && txn.value == 1
-            })
+            .filter(|txn| is_valid_ready_txn(txn, &block_window, &genesis_config_hex))
             .collect();
         filtered_txns
     } else {
         Vec::new()
     }
 }
 
+/// Checks if the provided transaction meets the criteria in order to be
+/// considered a valid ready-transaction
+fn is_valid_ready_txn(
+    txn: &TransactionDetails,
+    block_window: &Range<u32>,
+    genesis_config_hex: &String,
```

Comment:
I think I'd prefer to clarify this is a hash rather than clarifying the representation which I think we can assume it would be in `hex` given the type.
```suggestion
    genesis_config_hash: &String,
```

Reply:
I think it would actually be better to take the hash type here, not a string. Then you don't need to cast to hex when calling this method, only inside this method when comparing to the tx data.

Reply:
> I think it would actually be better to take the hash type here, not a string. Then you don't need to cast to hex when calling this method, only inside this method when comparing to the tx data.

But then we would be calling `.to_hex()` every single time for every transaction we check